### DISCUSSION TITLE
Finding current version used for the Mend Renovate hosted app on github.com?

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#25913] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### What is your question?

A Renovate hosted app user is asking me:

> @HonkingGoose, we do use the version hosted by Mend, any pointer to check the version being used?

And I don't know the correct answer, or at least the _easiest_ answer.

I tried these links:

- [Renovate docs, troubleshooting](https://docs.renovatebot.com/troubleshooting/)
- [Renovate docs, FAQ when is the mend app updated](https://docs.renovatebot.com/faq/#when-is-the-mend-renovate-app-updated-with-new-renovate-versions)

The Renovate debug logs that you can find on the "Mend dashboard" shows the Renovate version. I remember `@rarkins` saying somewhere that the PR footer of Renovate shows the version, but I checked a [recent Renovate PR on my `git-gosling` repository](https://github.com/HonkingGoose/git-gosling/pull/616) and it doesn't mention the Renovate version, here's the PR footer text from the linked PR:

```
This PR has been generated by [Mend Renovate](https://www.mend.io/free-developer-tools/renovate/). View repository job log [here](https://developer.mend.io/github/HonkingGoose/git-gosling).
<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNy41OS44IiwidXBkYXRlZEluVmVyIjoiMzcuNTkuOCIsInRhcmdldEJyYW5jaCI6Im1haW4ifQ==-->
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by secustor** [comment]
The debug log is base64 encoded. 

If you decode you get the the information you want: 
```shell
echo eyJjcmVhdGVkSW5WZXIiOiIzNy41OS44IiwidXBkYXRlZEluVmVyIjoiMzcuNTkuOCIsInRhcmdldEJyYW5jaCI6Im1haW4ifQ== | base64 --decode
{"createdInVer":"37.59.8","updatedInVer":"37.59.8","targetBranch":"main"}
```

**Reply by HonkingGoose** [reply]
@rarkins Why do we "hide" the Renovate version number in the PR footer by base64-encoding it? It would be easier to see the version in a plain format.

**Reply by rarkins** [reply]
It's not solving the same problem.

We embed Renovate versions in case something's gone wrong, and we need to debug. We try to minimize the space used so that we don't cannibalize the space for release notes, etc.

If someone wants to know the latest version running on the hosted app, it's to be found in the logs and not embedded in the PR. We don't update the PR every time we run with a newer version - only if we _made a change_, otherwise it would be too noisy and use up API calls unnecessarily.

**Reply by HonkingGoose** [reply]
> We try to minimize the space used so that we don't cannibalize the space for release notes, etc.

I must be missing something here, but are we _actually_ minimizing the space used? Here's the base64-encoded string:

```markdown
<!--renovate-debug:eyJjcmVhdGVkSW5WZXIiOiIzNy41OS44IiwidXBkYXRlZEluVmVyIjoiMzcuNTkuOCIsInRhcmdldEJyYW5jaCI6Im1haW4ifQ==-->
```

And the _expanded_ string:

```markdown
<!--renovate-debug:{"createdInVer":"37.59.8","updatedInVer":"37.59.8","targetBranch":"main"}-->
```

The Base64 string is _longer_ than the "stuff JSON between the debug markers" string.

I think we care about the number of characters that Renovate uses in it's PR body? So why not use the _expanded_ string instead? :upside_down_face:

**Reply by rarkins** [reply]
We should look into that

**Reply by rarkins** [reply, marked as answer]
#26412
